### ISSUE TITLE
Why might `gles` not work on my GeForce GTS 450?

### ISSUE THREAD

**Issue by Des-Nerger** [gfx-rs/wgpu#3820]
Why might `gles` not work on my Nvidia GeForce GTS 450? How can I investigate this?
I'm on Linux. When I run wgpu-powered frameworks, my GPU was ignored, choosing `Mesa (LLVM)` instead. When I tried to select the gles+GPU combo explicitly through environment variables, the app just didn't start.

Here's an output from `$ es2_info`: [es2_info.log](https://github.com/gfx-rs/wgpu/files/11608746/es2_info.log)

**Comment by cwfitzgerald** [comment]
What is the output of `wgpu-info` available on this repo?

**Comment by Des-Nerger** [comment]
[gts450_wgpu-info.log](https://github.com/gfx-rs/wgpu/files/11633400/gts450_wgpu-info.log)

**Comment by cwfitzgerald** [comment]
Ah, it seems like you're falling victim to https://github.com/gfx-rs/wgpu/issues/3792 plus our decision to take _any_ vulkan device over a gl device even if it's a software implementation. We should consider changing the logic to be a bit more logical, but we won't be able to "make the right choice" on all machines.

**Comment by Des-Nerger** [comment]
Wait, but if I run, let's say `$ WGPU_ADAPTER_NAME=450 cargo run --release --example cube`, I get this:
```
Using GeForce GTS 450/PCIe/SSE2 (Gl)
thread 'main' panicked at 'Surface isn't supported by the adapter.', wgpu/examples/cube/../framework.rs:276:22
```
Also, should I worry about `Compliant: false` in the `gts450_wgpu-info.log` I've posted above?

**Comment by cwfitzgerald** [comment]
No - that's just saying it's not fully WebGPU compliant, which is expected as WebGPU targets vulkan/metal/dx12, not GL.

Run with `RUST_LOG=info` in your env, and can you post that here?

**Comment by Des-Nerger** [comment]
[gts450-RUST_LOG=info.txt](https://github.com/gfx-rs/wgpu/files/11702682/gts450-RUST_LOG.info.txt)

**Comment by Des-Nerger** [comment]
Any idea what might cause that? From what I can hypothesize, it's the Nvidia EGL implementation for GTS 450 that is lacking some features the wgpu's gles backend needs.

**Comment by Des-Nerger** [comment]
Ok, I was able to fix it for myself using ANGLE:

- Downloaded the latest Chrome browser.
- Extracted `libEGL.so` and `libGLESv2.so` from it.
- Created the symlinks:
```
$ ln -s libEGL.so libEGL.so.1
$ ln -s libGLESv2.so libGLESv2.so.2
```
- And then run the `cube` example like this:
```
$ LD_LIBRARY_PATH=/path/to/those/ANGLE/libs WGPU_ADAPTER_NAME=450 cargo run --release --example cube
```
With that, I got around 3.5 times smaller `avg frame time` ( ~0.85ms now (with `ANGLE (NVIDIA Corporation, GeForce GTS 450/PCIe/SSE2, OpenGL 4.5.0 NVIDIA 390.157) (Gl)`), vs. ~3.0ms before (with `llvmpipe (LLVM 15.0.7, 256 bits) (Vulkan)`) )
